lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -159,16 +160,43 @@ def setup(self):
 
         self.log.debug(f"Output DL3 file: {self.output_file}")
 
+        try:
+            with fits.open(self.input_irf) as hdul:
+                self.use_energy_dependent_cuts = (
+                    "GH_CUT" not in hdul["EFFECTIVE AREA"].header
+                )
+        except KeyError:
+            raise ToolConfigurationError(
+                f"{self.input_irf} does not have GH CUTS HDU, "
+                "the energy-dependent gammaness cuts HDU, or "
+                " any global cut information in the Header value"
+            )
+
     def start(self):
 
         self.data = read_data_dl2_to_QTable(str(self.input_dl2))
-        ## To reduce the table columns further, add a selection of columns to be read and used.
         self.effective_time, self.elapsed_time = get_effective_time(self.data)
         self.run_number = run_info_from_filename(self.input_dl2)[1]
 
         self.data = self.event_sel.filter_cut(self.data)
-        self.data = self.fixed_cuts.gh_cut(self.data)
-        self.data = add_icrs_position_params(self.data, self.source_pos)
+
+        if self.use_energy_dependent_cuts:
+            self.energy_dependent_gh_cuts = QTable.read(
+                self.input_irf, hdu="GH_CUTS"
+            )
+
+            self.data = self.cuts.apply_energy_dependent_gh_cuts(
+                self.data, self.energy_dependent_gh_cuts
+            )
+            self.data = add_icrs_position_params(self.data, self.source_pos)
+            self.log.info(
+                "Using gamma efficiency of "
+                f'{self.energy_dependent_gh_cuts.meta["GH_EFF"]}'
+            )
+        else:
+            self.data = self.cuts.apply_global_gh_cut(self.data)
```

Comment:
Does this use the `GH_CUT` value in the IRF header or does this use whatever is in the default config? 

It should use the `GH_CUT` value from the IRF.

Reply:
Ah yes! Thanks for spotting this